rnd/autogpt_server/autogpt_server/server/ws_api.py

```diff
@@ -1,33 +1,89 @@
-from fastapi import WebSocket, WebSocketDisconnect
+import asyncio
+import logging
 
+from autogpt_libs.auth import auth_middleware, parse_jwt_token
+from fastapi import Depends, FastAPI, HTTPException, WebSocket, WebSocketDisconnect
+from fastapi.middleware.cors import CORSMiddleware
+
+from autogpt_server.data.queue import AsyncRedisEventQueue
+from autogpt_server.data.user import DEFAULT_USER_ID
 from autogpt_server.server.conn_manager import ConnectionManager
 from autogpt_server.server.model import ExecutionSubscription, Methods, WsMessage
+from autogpt_server.util.settings import Settings
 
+settings = Settings()
 
-async def websocket_router(websocket: WebSocket, manager: ConnectionManager):
-    await manager.connect(websocket)
-    try:
-        while True:
-            data = await websocket.receive_text()
-            message = WsMessage.model_validate_json(data)
-            if message.method == Methods.SUBSCRIBE:
-                await handle_subscribe(websocket, manager, message)
+app = FastAPI()
+event_queue = AsyncRedisEventQueue()
+_connection_manager = None
 
-            elif message.method == Methods.UNSUBSCRIBE:
-                await handle_unsubscribe(websocket, manager, message)
-            else:
-                print("Message type is not processed by the server")
-                await websocket.send_text(
-                    WsMessage(
-                        method=Methods.ERROR,
-                        success=False,
-                        error="Message type is not processed by the server",
-                    ).model_dump_json()
-                )
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=[
+        "*",
+    ],
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+
+def get_connection_manager():
+    global _connection_manager
+    if _connection_manager is None:
+        _connection_manager = ConnectionManager()
+    return _connection_manager
+
+
+@app.on_event("startup")
+async def startup_event():
+    await event_queue.connect()
+    manager = get_connection_manager()
+    asyncio.create_task(event_broadcaster(manager))
 
-    except WebSocketDisconnect:
-        manager.disconnect(websocket)
-        print("Client Disconnected")
+
+@app.on_event("shutdown")
+async def shutdown_event():
+    await event_queue.close()
+
+
+async def event_broadcaster(manager: ConnectionManager):
+    while True:
+        event = await event_queue.get()
+        if event is not None:
+            await manager.send_execution_result(event)
+
+
+def get_user_id(payload: dict = Depends(auth_middleware)) -> str:
+    if not payload:
+        # This handles the case when authentication is disabled
+        return DEFAULT_USER_ID
+
+    user_id = payload.get("sub")
+    if not user_id:
+        raise HTTPException(status_code=401, detail="User ID not found in token")
+    return user_id
```

Comment:
Copy-pasted from `.rest_api` -> better to move into a shared `.utils`

Reply:
this isn't used here actually, removed it and put it in utils regardless